src/Database/PG/Query/Pool.hs

```diff
@@ -137,6 +137,14 @@ class FromPGTxErr e where
 class FromPGConnErr e where
   fromPGConnErr :: PGConnErr -> e
 
+-- These instances permit (FromPGConnErr e, MonadError e m)=> ... to unify with IO
+instance FromPGTxErr IOException where
+  fromPGTxErr = userError . show
```

Comment:
My only small concern is that `show` here isn't going to result in a very useful error message if this ends up making its way into logs or into the UI. Any thoughts?

Reply:
Right, I think they'll be a little uglier (containing escaped JSON)
In practice, with the upcoming changes to graphql-engine it will mean we'll get these uglier log messages produced from `forkImmortal`'s logging/restarting machinery, on the rare(?) occasion that PG throws an error of some kind to the `LISTEN` thread (or in other cases where another future change makes use of these instances)

The benefit is we can remove https://github.com/hasura/graphql-engine-mono/blob/main/server/src-lib/Hasura/Server/SchemaUpdate.hs#L213-L221
And instead let `forkImmortal` handle restarting and logging (which it's doing anyway)

...hm, but actually maybe `forkImmortal` can be re-written to be in `ToJSON e, MonadError e m` so we get structured logging...

Reply:
@jberryman Just an FYI, we have a branch we are hoping to merge soon that removes the use of `listen` you linked and replaces it with polling in the OSS engine.

Reply:
@ssbothwell ah, thanks for the heads up. Can you link me to that?

Reply:
nevermind, https://github.com/hasura/graphql-engine-mono/pull/738/files

Reply:
yes thats the one!